### ISSUE TITLE
Persist DataFn should accept TagIterator to avoid unnecessary reconversions

### ISSUE THREAD

**Issue by justinjc** [m3db/m3#1675]
Currently, the [DataFn](https://github.com/m3db/m3/blob/c9820911fadc15f64c3af2ee8a07a8cb429d5374/src/dbnode/persist/types.go#L36) accepts `ident.Tags`. There are scenarios when we already have an `ident.TagIterator`, such as on [`Read()`](https://github.com/m3db/m3/blob/c9820911fadc15f64c3af2ee8a07a8cb429d5374/src/dbnode/persist/fs/read.go#L309), meaning it has to be unpacked to be passed into this function. This is wasteful because internally, these `ident.Tags` are reconverted into an `ident.TagIterator` before use.

tl;dr: instead of `tagsIter -> tags -> tagsIter`, just pass `tagsIter` through.
